File: 739. Daily Temperatures.py
class Solution:
    def dailyTemperatures(self, T: List[int]) -> List[int]:
        stk, ans = [], [0]*len(T)
        
        for cur, num in enumerate(T):
            while stk and num>stk[-1][1]:
                p, p_temp = stk.pop()
                ans[p] = cur-p
            
            stk.append((cur, num))
        
        return ans

File: 739. Daily Temperatures.js
/**
 * @param {number[]} T
 * @return {number[]}
 */
var dailyTemperatures = function(T) {
    let [stk, ans] = [[], new Array(T.length)];
    for(let cur=0;cur<T.length;cur++){
        ans[cur] = 0;

        while(stk.length!=0 && T[cur]>stk.slice(-1)[0][1]){
            let [p, p_temp] = stk.pop();
            ans[p] = cur-p;
        }
        stk.push([cur, T[cur]]);
    }
    return ans;
};